pulp_container/app/tasks/sync_stages.py

```diff
@@ -459,33 +488,36 @@ async def create_signatures(self, man_dc, signature_source):
                     signature_raw = f.read()
 
                 signature_counter += 1
-                signature_json = extract_data_from_signature(signature_raw, man_dc.content.digest)
-                if signature_json is None:
-                    continue
-
-                sig_digest = hashlib.sha256(signature_raw).hexdigest()
-                signature = ManifestSignature(
-                    name=f"{man_dc.content.digest}@{sig_digest[:32]}",
-                    digest=f"sha256:{sig_digest}",
-                    type=SIGNATURE_TYPE.ATOMIC_SHORT,
-                    key_id=signature_json["signing_key_id"],
-                    timestamp=signature_json["signature_timestamp"],
-                    creator=signature_json["optional"].get("creator"),
-                    data=base64.b64encode(signature_raw).decode(),
-                )
-                sig_dc = DeclarativeContent(
-                    content=signature,
-                    extra_data={"signed_manifest_dc": man_dc},
-                )
-                signature_dcs.append(sig_dc)
-
-            return signature_dcs
+                sig_dc = self._create_signature_declarative_content(signature_raw, man_dc)
+                if sig_dc:
+                    signature_dcs.append(sig_dc)
 
         elif signature_source == SIGNATURE_SOURCE.API_EXTENSION:
-            # TODO in a PR for the extension support
-            pass
+            signatures_url = urlpath_sanitize(
+                self.remote.url,
+                "extensions/v2",
+                self.remote.upstream_name,
+                "signatures",
+                man_dc.content.digest,
+            )
+            signatures_downloader = self.remote.get_downloader(url=signatures_url)
+            await signatures_downloader.run()
+            with open(signatures_downloader.path) as signatures_fd:
+                api_extension_signatures = json.loads(signatures_fd.read())
+            for signature in api_extension_signatures.get("signatures", []):
+                if (
+                    signature["schemaVersion"] == SIGNATURE_API_EXTENSION_VERSION
+                    and signature["type"] == SIGNATURE_TYPE.ATOMIC_SHORT
+                ):
+                    signature_base64 = signature["content"]
+                    signature_raw = base64.b64decode(signature_base64)
+                    sig_dc = self._create_signature_declarative_content(
+                        signature_raw, man_dc, signature["name"], signature_base64
```

Comment:
nitpick, while i was reviewing Lubos'es pr -  we should assert that `name` is part of the provided body structure

Reply:
or `signature.get('name')